src/xrpld/app/tx/detail/VaultWithdraw.cpp

```diff
@@ -307,11 +314,57 @@ VaultWithdraw::doApply()
         // else quietly ignore, account balance is not zero
     }
 
-    auto const dstAcct = [&]() -> AccountID {
-        if (ctx_.tx.isFieldPresent(sfDestination))
-            return ctx_.tx.getAccountID(sfDestination);
-        return account_;
-    }();
+    auto const dstAcct = ctx_.tx[~sfDestination].value_or(account_);
+
+    if (!vaultAsset.native() && dstAcct != vaultAsset.getIssuer() &&
+        dstAcct == account_)
+    {
+        if (auto const ter = std::visit(
+                [&]<ValidIssueType TIss>(TIss const& issue) -> TER {
+                    if constexpr (std::is_same_v<TIss, Issue>)
+                    {
+                        std::shared_ptr<SLE> dummy = {};
+                        if (auto const sleLine =
```

Comment:
This is more efficient:

     if (!view().exists(keylet::line(dstAcct, issue)))

Reply:
also more explicit - thanks !